Utilities/gdcmopenjpeg/src/lib/openjp2/j2k.c

```diff
@@ -6921,8 +6921,9 @@ OPJ_BOOL opj_j2k_setup_encoder(opj_j2k_t *p_j2k,
                                        image->comps[0].h * image->comps[0].prec) /
                                       ((double)parameters->tcp_rates[parameters->tcp_numlayers - 1] * 8 *
                                        image->comps[0].dx * image->comps[0].dy));
-            if (temp_size > (OPJ_FLOAT32)INT_MAX) {
-                parameters->max_cs_size = (OPJ_FLOAT32)INT_MAX;
+            static const OPJ_FLOAT32 max_float_size_allowed = (1 << FLT_MANT_DIG); // LARGEST INTEGER STORABLE IN FLOAT
+            if (temp_size > max_float_size_allowed) {
+                parameters->max_cs_size = max_float_size_allowed;
             } else {
```

Comment:
FYI, the warning
```
GDCM/Utilities/gdcmopenjpeg/src/lib/openjp2/j2k.c:6925:43: warning: overflow in implicit constant conversion [-Woverflow]
                 parameters->max_cs_size = (OPJ_FLOAT32)INT_MAX;
```
was fixed in _openjpeg_ **2.3.1**:  C-cast (OPJ_FLOAT32) was simply removed

**2.3.0**
```

            if (temp_size > (OPJ_FLOAT32)INT_MAX) {
                parameters->max_cs_size = (OPJ_FLOAT32)INT_MAX;
            } else {
                parameters->max_cs_size = (int) floor(temp_size);
            }
```

**2.3.1**

```
            if (temp_size > INT_MAX) {
                parameters->max_cs_size = INT_MAX;
            } else {
                parameters->max_cs_size = (int) floor(temp_size);
            }
```

Seems to be actual in _openjpeg_ master
https://github.com/uclouvain/openjpeg/blob/0782cd075d7e9678bf1cb448c1f3d048fabe4489/src/lib/openjp2/j2k.c#L7675